src/ble/BtpEngine.cpp

```diff
@@ -403,7 +403,7 @@ CHIP_ERROR BtpEngine::HandleCharacteristicReceived(System::PacketBufferHandle &&
         }
         if (!mRxBuf.IsNull())
         {
-            ChipLogError(Ble, "With rx buf data length = %u", mRxBuf->DataLength());
+            ChipLogError(Ble, "With rx buf data length = %u", static_cast<uint16_t>(mRxBuf->DataLength()));
```

Comment:
%u is unsigned. Cast to unsigned.

Reply:
Instead of casting to uint16_t for printing,  use the format specifiers defined in <inttypes.h> for right fixed-width integers. See example above.

Reply:
Can go either way, i.e., unsigned or PRIu16.